Configure gradient data requirement automatically (#1371)
precice/precice: docs/changelog/1371.md

> - Removed the explicit gradient data flag (`<data: ... gradient="on" />`) for nearest-neighbor-gradient mapping. The gradient data requirement will be deduced automaticall by preCICE.

Reviewer: ```suggestion
- Removed the explicit gradient data flag (`<data: ... gradient="on" />`) for nearest-neighbor-gradient mapping. The gradient data requirement is now deduced automatically by preCICE.
```
